applications/webapp/src/Pages/User/userPage.tsx

```diff
@@ -22,21 +22,44 @@ export default function UserPage() {
   const theme = useTheme();
   const style = styles(theme).userPage;
 
+  const [enabled, setEnabled] = useState(true);
+  const [elevation, setElevation] = useState(0);
+
+  const editUser = () => {
+    const elevate = 5;
+    const zeroElevation = 0;
+    setEnabled((enabled) => !enabled);
+    setElevation((elevation) =>
+      elevation === zeroElevation ? elevate : zeroElevation,
+    );
+  };
+
+  const { data } = useQuery(getUserQuery);
+  const { getUser } = data || {};
+
   const mockUser = {
     phonenumber: "123456789",
     address: "Addis Ababa",
     birthDate: "April 19 2001",
-    description: "Hi I am Yonas",
+    description: `Hi I am ${getUser?.username}`,
   };
 
-  const { data } = useQuery(getUserQuery);
-
-  const { getUser } = data || {};
+  const userDetail = {
+    username: getUser?.username,
+    firstName: getUser?.firstName,
+    lastName: getUser?.lastName,
+    email: getUser?.email,
+    phoneNumber: mockUser["phonenumber"],
+    address: mockUser.address,
+    birthdate: mockUser.birthDate,
+  };
 
-  const Username = getUser?.username;
-  const FirstName = getUser?.firstName;
-  const LastName = getUser?.lastName;
-  const Email = getUser?.email;
+  const card = {
+    editable: enabled,
+    setEditable: setEnabled,
+    elevation: elevation,
+    setElevation: setElevation,
```

Comment:
This is a bad practice. Very tempting, yes, because it's simple. But bad. It is bad because it breaks isolation. State is by default an internal, private life of a component. What happens here is that you expose the state management to external component. Normally this should not happen. In React, if the state of component must depend on external factors then there are two ways:
- component props. we pass new props, component gets updated
- context or other high-level state-management tool like Redux.

Try to reconsider this approach and find a different approach. Example: pass a callback function to underlying component like "onUpdateStateChanged". That component can call it and once it is called it is executed in the context of this components and updates its state. 

Crude example
```typescript
function ComponentA() {
  const [val, setVal] = useState(0);
  const onValChanged = (newVal) => {
    setVal(newVal);
  }
  
  return (
    <ComponentB onChange={onValChanged} />
  )
}
```

Reply:
However, in this particular case, perhaps, context would be a better approach.